pkg/controller/namespace/namespace_controller.go

```diff
@@ -136,7 +136,7 @@ func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Res
 				log.WithError(err).Error("failed to get the available Jaeger pods")
 				return reconcile.Result{}, tracing.HandleError(err, span)
 			}
-
+			patch := client.MergeFrom(dep.DeepCopy())
```

Comment:
Can it be moved closer to where it's being used? Also, you might not need `dep.DeepCopy()`

Reply:
Not sure I don't need the deepcopy, the thing is, the structure could have some other nested structures as a pointers. and if I just use dep the tests doesn't pass :(